Combine user edit and password change forms

helenst opened the pull request:
They are now controlled by a single button - this is to prevent users
from accidentally not saving their user data due to not realising saves
must be done separately.

The two forms remain separate (as they are based on Django forms) but
the view will treat them atomically - data will be saved on both forms
if and only if they are both correctly filled in. (Note that in the
case of the password form, correctly filled in can mean empty - the user
may choose not to change their password).

Tests also added.

helenst committed:
Combine user edit and password change forms

They are now controlled by a single button - this is to prevent users
from accidentally not saving their user data due to not realising saves
must be done separately.

The two forms remain separate (as they are based on Django forms) but
the view will treat them atomically - data will be saved on both forms
if and only if they are both correctly filled in. (Note that in the
case of the password form, correctly filled in can mean empty - the user
may choose not to change their password).

Tests also added.

Hwesta reviewed (approved):
Thank you for the contribution, this look great!  Especially pleased to see tests for the form & view, since I've been unsure how to do this and this example is very clear.

scollazo commented:
Please Jenkins, build this. Thanks

scollazo commented:
test this please

helenst committed:
Use setUpTestData to prevent data clashes

setUpClass does not tear data down automatically, so there
were duplicate data problems across multiple test cases.

sevein commented:
Thank you so much!
